### PR TITLE
[manager] Add `DetachAndDeleteHook` method to the manager

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR introduces a new method to the manager: `DetachAndDeleteHook(id ProbeIdentificationPair)`. The goal of this method is to provide a way to override the deletion decision of `DetachHook(id ProbeIdentificationPair)` without breaking the existing API.

### Motivation

Providing the ability to stop and remove a probe from the manager is useful in situation where you want to get rid of a Probe that you no longer need and won't need in the future.

### CODE REVIEW

File: manager.go [hunk 1]
```diff
@@ -909,6 +910,27 @@ func (m *Manager) DetachHook(id ProbeIdentificationPair) error {
...
 
+// DetachAndDeleteHook - Detaches an eBPF program from a hook point and delete it from the manager.
+func (m *Manager) DetachAndDeleteHook(id ProbeIdentificationPair) error {
+	idToDelete := -1
+	var err error
+	for mID, mProbe := range m.Probes {
+		if mProbe.Matches(id) {
+			if err = mProbe.Stop(); err != nil {
+				return fmt.Errorf("couldn't stop probe %s: %w", id, err)
+			}
```
Comment (brycekahle) [inline manager.go:921]: Is this missing a `mProbe.Detach` call, like the `Detach` function?

**Comment by brycekahle** [comment]
What is the difference between the new function and the existing one? They seem to do the same thing?

File: manager.go (hunk 1 above)
Comment (Gui774ume) [reply manager.go:919]: Calling `mProbe.Stop()` does the same thing as `mProbe.Detach()` + it closes the underlying eBPF program. So in this specific case it should be enough.

**Comment by Gui774ume** [comment]
> What is the difference between the new function and the existing one? They seem to do the same thing?

We changed the way we load and clone eBPF TC classifiers so you're right, we don't need this function anymore. (The initial goal of having this function was to force a delete even when there was only one instance of a TC program in kernel space ... but since we now load our TC egress and ingress program by default, without attaching them, there will always be at least one instance of them, thus deleting new ones with `DetachHook()` will always close the underlying programs).

TL;DR this is no longer needed, closing this PR 👍🏻
